app/models/user_system_notification.rb

```diff
@@ -4,8 +4,14 @@ class UserSystemNotification < ApplicationRecord
   belongs_to :user
   belongs_to :system_notification
 
+  after_create :send_email
```

Comment:
Hmm.. Yeap. Actually, I was trying to avoid sending mail on callback because I think this action should be called explicitly. But I didn't find a better place to call it. 
I will change it to callback with proc.

Reply:
@Ducz0r approach is IMO better, you can see at once when this callback is called.